engine/collisionmanager.cpp

```diff
@@ -9,16 +10,26 @@
 namespace pong
 {
 
-void CollisionManager::ProcessCollisions(const std::vector<std::unique_ptr<GameObject>>& gameObjects)
+void CollisionManager::ProcessCollisions(const std::vector<std::unique_ptr<ColliderBox>>& colliders)
 {
-    for (int index = 0; index < gameObjects.size(); index++)
+    for (int index = 0; index < colliders.size(); index++)
     {
-        auto& gameObject = gameObjects[index];
-        for (int otherIndex = index + 1; otherIndex < gameObjects.size(); otherIndex++)
+        auto& collider = colliders[index];
+        if (collider->GetGameObject()->GetComponent<Transform>() != nullptr)
```

Comment:
I would consider using an initializer in the if statement here because you use this so frequently in the block


I believe the syntax is something like
```
if (auto* transform = collider->GetGameObject()->GetComponent<Transform>(); transform != nullptr)
{
                collider->OnPositionUpdate(transform->mPosition);
...    
}
```